pygeos/coordinates.py

```diff
@@ -4,19 +4,26 @@
 __all__ = ["apply", "count_coordinates", "get_coordinates", "set_coordinates"]
 
 
-def apply(geometry, transformation):
+def apply(geometry, transformation, include_z=False):
     """Returns a copy of a geometry array with a function applied to its
     coordinates.
 
-    All returned geometries will be two-dimensional; the third dimension will
-    be discarded, if present.
+    With the default of ``include_z=False``, all returned geometries will be
```

Comment:
Maybe `include_z` should better be `True`? I don't care too much about breaking the API at this stage of pygeos.

Reply:
I should maybe check the performance overhead in case all your geometries are actually 2D (which I think is by far the more common case) and when using ``include_z=True`` 
And note that this *does* have an influence on the function being passed, because this function will now by default have to deal with z coordinates

Reply:
Hmm, doesn't seem to matter much in case of an array of points:

```
In [3]: arr = pygeos.points(np.random.randn(1_000_000, 2))

In [4]: 

In [4]: arr
Out[4]: 
array([<shapely.geometry.Point POINT (1.459720029804123 -1.660916227958426) >,
       <shapely.geometry.Point POINT (-2.336782731229145 2.129473193643444) >,
       <shapely.geometry.Point POINT (0.1039792330580081 -2.048775667056068) >,
       ...,
       <shapely.geometry.Point POINT (0.2416382706879835 -1.697157520118229) >,
       <shapely.geometry.Point POINT (0.2381751392103192 -0.05537962788130606) >,
       <shapely.geometry.Point POINT (-0.3401224146947915 0.2503277994539527) >],
      dtype=object)

In [5]: pygeos.apply(arr, lambda coords: coords*2)
Out[5]: 
array([<shapely.geometry.Point POINT (2.919440059608247 -3.321832455916851) >,
       <shapely.geometry.Point POINT (-4.673565462458289 4.258946387286889) >,
       <shapely.geometry.Point POINT (0.2079584661160162 -4.097551334112135) >,
       ...,
       <shapely.geometry.Point POINT (0.483276541375967 -3.394315040236458) >,
       <shapely.geometry.Point POINT (0.4763502784206384 -0.1107592557626121) >,
       <shapely.geometry.Point POINT (-0.680244829389583 0.5006555989079053) >],
      dtype=object)

In [6]: %timeit pygeos.apply(arr, lambda coords: coords*2)
472 ms ± 4.88 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [7]: %timeit pygeos.apply(arr, lambda coords: coords*2, include_z=True)
472 ms ± 2.62 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Reply:
Another default could also be to "infer" the dimensionality, either by checking all geometries (`pygeos.has_z` is quite cheap compared to `pygeos.apply`, it's 16ms for the example above, where the dummy apply takes 470ms) or eg the first geometry. And then the user can avoid this inference by either passing include_z=True/False

But given that it doesn't seem to give much overhead (and actually less that doing a `pygeos.has_z` ...), I am also fine with simply having `include_z=True` as the default. We only need to be aware that the applied function will then always receive a (N, 3) array as argument by default, and not a (N, 2)

Reply:
Personally, I prefer the default of `include_z=False`.  Since 2D is much more common, it makes sense to make 3D opt-in, whereas it would be unexpected to have to suppress 3D by always passing `include_z=False`.

Reply:
Yeah I agree, the argument of having to provide a `(N, 3)` -> `(N, 3) function is convincing